tests/engine/unit/segments/segments_model.test.ts

```diff
@@ -62,11 +62,32 @@ const conditionMatchCases: [string, string | number | boolean, string, boolean][
     [CONDITION_OPERATORS.NOT_CONTAINS, 'bar', 'baz', true],
     [CONDITION_OPERATORS.REGEX, 'foo', '[a-z]+', true],
     [CONDITION_OPERATORS.REGEX, 'FOO', '[a-z]+', false],
+    [CONDITION_OPERATORS.EQUAL, "1.0.0", "1.0.0:semver", true],
+    [CONDITION_OPERATORS.EQUAL, "1.0.0", "1.0.0:semver", true],
+    [CONDITION_OPERATORS.EQUAL, "1.0.0", "1.0.1:semver", false],
+    [CONDITION_OPERATORS.NOT_EQUAL, "1.0.0", "1.0.0:semver", false],
+    [CONDITION_OPERATORS.NOT_EQUAL, "1.0.0", "1.0.1:semver", true],
+    [CONDITION_OPERATORS.GREATER_THAN, "1.0.1", "1.0.0:semver", true],
+    [CONDITION_OPERATORS.GREATER_THAN, "1.0.0", "1.0.0-beta:semver", true],
+    [CONDITION_OPERATORS.GREATER_THAN, "1.0.1", "1.2.0:semver", false],
+    [CONDITION_OPERATORS.GREATER_THAN, "1.0.1", "1.0.1:semver", false],
+    [CONDITION_OPERATORS.GREATER_THAN, "1.2.4", "1.2.3-pre.2+build.4:semver", true],
+    [CONDITION_OPERATORS.LESS_THAN, "1.0.0", "1.0.1:semver", true],
+    [CONDITION_OPERATORS.LESS_THAN, "1.0.0", "1.0.0:semver", false],
+    [CONDITION_OPERATORS.LESS_THAN, "1.0.1", "1.0.0:semver", false],
+    [CONDITION_OPERATORS.LESS_THAN, "1.0.0-rc.2", "1.0.0-rc.3:semver", true],
+    [CONDITION_OPERATORS.GREATER_THAN_INCLUSIVE, "1.0.1", "1.0.0:semver", true],
+    [CONDITION_OPERATORS.GREATER_THAN_INCLUSIVE, "1.0.1", "1.2.0:semver", false],
+    [CONDITION_OPERATORS.GREATER_THAN_INCLUSIVE, "1.0.1", "1.0.1:semver", true],
+    [CONDITION_OPERATORS.LESS_THAN_INCLUSIVE, "1.0.0", "1.0.1:semver", true],
+    [CONDITION_OPERATORS.LESS_THAN_INCLUSIVE, "1.0.0", "1.0.0:semver", true],
+    [CONDITION_OPERATORS.LESS_THAN_INCLUSIVE, "1.0.1", "1.0.0:semver", false],
     ['BAD_OP', 'a', 'a', false]
 ];
 
 test('test_segment_condition_matches_trait_value', () => {
     for (const testCase of conditionMatchCases) {
+        console.log(testCase);
```

Comment:
I think this can be removed?

Reply:
@yuriihorodnyi21 I went ahead and did this myself so I can get this merged. Thanks for the PR 👍